Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp

```diff
@@ -98,6 +100,14 @@ static void EnqueueEvent(u64 userdata, s64 cycles_late = 0)
   Update();
 }
 
+void SDIO_EventNotify_CPUThread(u64 userdata, s64 cycles_late)
+{
+  auto device =
+      static_cast<CWII_IPC_HLE_Device_sdio_slot0*>(GetDeviceByName("/dev/sdio/slot0").get());
```

Comment:
No, that's a bad idea.

std::static_pointer_cast creates a new shared_ptr, which increments the atomic counter. And since it's an atomic operation it, the compiler can't optimised out the increment then decrement.

This is better.